### PR TITLE
Issue #SB-0000 feat : Implementation of Play - Pause and Normal screen for Interactive Content.

### PR DESCRIPTION
**Pull request by Xitija**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/sunbird-video-player/package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "@project-sunbird/sunbird-video-player-v9",
-  "version": "4.10.1",
+  "version": "4.10.2",
   "peerDependencies": {
     "@angular/common": ">= 8.0.0",
     "@angular/core": ">= 8.0.0",
     "@types/video.js": "7.5.5",
-    "video.js": "7.4.1",
```
Comment (HarishGangula) [inline projects/sunbird-video-player/package.json:8]: What is the reason to change the video js version and is it tested with the existing video plugins?

File: projects/sunbird-video-player/package.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "@project-sunbird/sunbird-video-player-v9",
-  "version": "4.10.1",
```
Comment (HarishGangula) [inline projects/sunbird-video-player/package.json:3]: Update according to the latest version

File: projects/sunbird-video-player/src/lib/components/video-player/video-player.component.ts [hunk 3]
```diff
@@ -13,8 +13,9 @@ import { ViewerService } from '../../services/viewer.service';
   styleUrls: ['./video-player.component.scss'],
   encapsulation: ViewEncapsulation.None
 })
-export class VideoPlayerComponent implements AfterViewInit, OnInit, OnDestroy {
+export class VideoPlayerComponent implements AfterViewInit, OnInit, OnDestroy, OnChanges {
   @Input() config: any;
+  @Input() action: string;
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/components/video-player/video-player.component.ts:18]: Can we make this an interface with action name and other properties,
**Example:**
Tomorrow if we get a use case like play from 1:45 min if answered all questions in question set or play from 1:10 min if unanswered

Also, please update the documents both README and [microsite](https://knowlg.sunbird.org/learn/product-and-developer-guide/player/v2/video-player) and add the unit test cases for the same/

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts [hunk 4]
```diff
@@ -217,6 +219,11 @@ export class SunbirdVideoPlayerComponent implements OnInit, AfterViewInit, OnDes
       }
       this.videoInstance.play();
       this.videoInstance.controls(true);
+      // if currently video is not in full screen and was previously full screen then set it back to full screen again
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:222]: Can you write the test cases for the same and it should not break existing flows in the portal and mobile apps, Can you confirm same?

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts [hunk 5]
```diff
@@ -227,6 +234,13 @@ export class SunbirdVideoPlayerComponent implements OnInit, AfterViewInit, OnDes
     this.QumlPlayerConfig.metadata['showEndPage'] = 'No';
     this.currentInterceptionTime = time;
     this.currentInterceptionUIId = identifier;
+    if (document.fullscreenElement) {
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:237]: Question set can also be played in full screen, i want to understand the reason for this changes

File: src/app/app.component.html [hunk 6]
```diff
@@ -1,8 +1,12 @@
 <div class="player-grid">
     <div>
         <div class="aspectratio" data-ratio="16:9" id="help-video-aspect-ratio" #aspectRatio>
-            <sunbird-video-player *ngIf="playerConfig" [playerConfig]="playerConfig" (playerEvent)="playerEvent($event)" (telemetryEvent)="telemetryEvent($event)"></sunbird-video-player>
-          </div>
+            <sunbird-video-player *ngIf="playerConfig" [playerConfig]="playerConfig" (playerEvent)="playerEvent($event)" (telemetryEvent)="telemetryEvent($event)" [action]="action"></sunbird-video-player>
```
Comment (HarishGangula) [inline src/app/app.component.html:4]: Make action an optional input field

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: projects/sunbird-video-player/package.json (hunk 1 above)
Comment (Xitija) [reply projects/sunbird-video-player/package.json:8]: When we were trying to implement the non-fullscreen mode for video player ,
Video fullscreen maximize and minimize via icon was not working 
We have tested plugins for resolution & speed .

File: projects/sunbird-video-player/package.json (hunk 2 above)
Comment (Xitija) [reply projects/sunbird-video-player/package.json]: Updated version to 4.10.3

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts (hunk 5 above)
Comment (Xitija) [reply projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:238]: For video fullscreen we are using video.js fullscreen so out quml player doesn't recognize the fullscreen and currently the Quml-player is not working thats why we are going with this approach.
In mobile application we launch the Interactive Video in landscape mode so even though video is in full screen or normal screen it doesnt affect the UI.

**Review by HarishGangula — CHANGES_REQUESTED** [review]
Still unit test cases and documentation needs to be updated in this PR

**Comment by HarishGangula** [comment]
@Xitija  Still the README file is out of date with your changes, Please update it

File: projects/sunbird-video-player/src/lib/components/video-player/video-player.component.ts (hunk 3 above)
Comment (Xitija) [reply projects/sunbird-video-player/src/lib/components/video-player/video-player.component.ts]: We have changed it to a property in an Interface , updated readme but I dont have access to microsite

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts (hunk 4 above)
Comment (Xitija) [reply projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:233]: In the mobile the Interactive Video is played in the landscape mode only so the fullscreen issue does not arise

Comment (HarishGangula) [reply projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:233]: Mobile app planning to implement portrait mode also, So keep that scenario also covered

**Review by HarishGangula — COMMENTED** [review]
There is one last review comment pending

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts (hunk 4 above)
Comment (Xitija) [reply projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:233]: https://github.com/Sunbird-Ed/Community/discussions/15
We previously had a discussion with the mobile team to open Interactive video in landscape mode only

**Review by HarishGangula — APPROVED** [review]

File: projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts (hunk 4 above)
Comment (Xitija) [reply projects/sunbird-video-player/src/lib/sunbird-video-player.component.ts:233]: Videos (without Interactive content / QuestionSet) can be played in both portrait and landscape mode.
But QuestionSet in the mobile app in portrait is not showing properly so whenever we have 
Interactive video having Questionset in it that is handled my mobile team to play it in landscape mode.
https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/pull/3016

File: projects/sunbird-video-player/package.json [hunk 7]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "@project-sunbird/sunbird-video-player-v9",
-  "version": "4.10.2",
```
Comment (HarishGangula) [inline projects/sunbird-video-player/package.json:3]: Please change version to 5.0.1

**Review by HarishGangula — CHANGES_REQUESTED** [review]

**Pull request merged**